File: backend/domain/entities/pedido.py
"""Entidade de domínio: Pedido com enum de status."""

import enum
import uuid
from dataclasses import dataclass, field
from datetime import datetime
from decimal import Decimal


class PedidoStatus(str, enum.Enum):
    pendente = "pendente"
    pago = "pago"
    rejected = "rejected"
    cancelled = "cancelled"
    in_process = "in_process"


@dataclass
class Pedido:
    planta_id: uuid.UUID
    email: str
    nome: str
    valor: Decimal
    id: uuid.UUID = field(default_factory=uuid.uuid4)
    telefone: str | None = None
    status: PedidoStatus = PedidoStatus.pendente
    mp_payment_id: str | None = None
    download_token: uuid.UUID | None = None
    expires_at: datetime | None = None
    criado_em: datetime = field(default_factory=datetime.utcnow)
    atualizado_em: datetime = field(default_factory=datetime.utcnow)

File: frontend/core/domain/entities/Pedido.ts
export interface PlainPedido {
  id: string;
  plantaId: string;
  email: string;
  nome: string;
  telefone: string | null;
  valor: number;
  status: 'pendente' | 'pago' | 'rejected' | 'cancelled' | 'in_process';
  mpPaymentId: string | null;
  downloadToken: string | null;
  expiresAt: string | null;
  criadoEm: string;
  atualizadoEm: string;
}

export class Pedido {
  constructor(
    public readonly id: string,
    public readonly plantaId: string,
    public readonly email: string,
    public readonly nome: string,
    public readonly telefone: string | null,
    public readonly valor: number,
    public readonly status: 'pendente' | 'pago' | 'rejected' | 'cancelled' | 'in_process',
    public readonly mpPaymentId: string | null,
    public readonly downloadToken: string | null,
    public readonly expiresAt: string | null,
    public readonly criadoEm: string,
    public readonly atualizadoEm: string
  ) {}
}
